Source/API/stochastic.h

```diff
@@ -18,7 +18,7 @@ class stochastic {
 
     virtual bool evaluate_probability(double probability) = 0;
 
-    virtual size_t weight_proportionate_selection(double* weights, size_t num_weights) = 0;
+    virtual size_t weight_proportionate_selection(const double* const weights, size_t num_weights) = 0;
```

Comment:
Interesting that you have marked this `double` as `const` but other `double`s have not been.

Reply:
That's because this one is an array so marking it const indicates the method won't change the content of the array. In the other cases it is simply a double and is passed by value so const doesn't make sense since the data is being copied.

Reply:
We should remove all opaque buffers. Can hold off until pointer audit.